performancecopilot/pcp
# mostly qa Changes committed to git@github.com:kmcdonell/pcp.git 20200110

Ken McDonell (13):
      qa/1423: extend sleeps
      qa/054: try to avoid "hang" in QA Farm
      qa/1155,11157,1170 and 1178: _not run for Ubuntu 19.10
      qa/1211: drop QAHOST filtering ... not needed
      qa/510: try to fix failures on RHEL/CentOS 6.10
      qa/1695: increase delay so valgrind can get started
      qa/admin/other-packages: changes for Ubuntu 19.10
      qa/check.callback.sample: make sure pmlogger failures are reported
      qa/667: various improvements
      qa/667: fix previous commit
      qa/src/test_pcp.python: don't use unicode strings with Python2
      qa/1661: foo.{a,b} is not Bourne shell syntax
      qa/1661.out: track cosmetic change in 1661

 qa/054                              |   12 +++
 qa/1155                             |   17 +++++
 qa/1157                             |   17 +++++
 qa/1170                             |   17 +++++
 qa/1178                             |   17 +++++
 qa/1211                             |    1 
 qa/1423                             |    9 ++
 qa/1661                             |   29 +++++++-
 qa/1661.out                         |    2 
 qa/1695                             |    2 
 qa/510                              |   44 +++++++++++--
 qa/667                              |  121 ++++++++++++++++++------------------
 qa/admin/other-packages/manifest    |    2 
 qa/admin/other-packages/unavailable |    1 
 qa/check.callback.sample            |    6 +
 qa/src/test_pcp.python              |    5 +
 16 files changed, 224 insertions(+), 78 deletions(-)

Details ...

commit 9c0cda827192dd28f1da26e65c69dc95a0d500e1
Author: Ken McDonell <kenj@kenj.id.au>
Date:   Thu Jan 23 17:00:37 2020 +1100

    qa/1661.out: track cosmetic change in 1661

commit 07d87683bc91ab6e51540befcf364603a2da4e71
Author: Ken McDonell <kenj@kenj.id.au>
Date:   Thu Jan 23 16:48:06 2020 +1100

    qa/1661: foo.{a,b} is not Bourne shell syntax
    
    This came from csh and has infected bash ... if we're really using
    /bin/sh, then foo.{a,b} is 'foo.{a,b}' not 'foo.a foo.b'.

commit 0769bc22341d5c7e56c4938864d9c220fe4124ea
Author: Ken McDonell <kenj@kenj.id.au>
Date:   Thu Jan 23 16:18:44 2020 +1100

    qa/src/test_pcp.python: don't use unicode strings with Python2
    
    Was failing on RHEL 6.10 (vm28) where Python 2.6.6 is the only
    Python installed.

commit c97cb07a352e9a7a099bedacdb781406a8e4ae25
Author: Ken McDonell <kenj@kenj.id.au>
Date:   Thu Jan 23 16:10:13 2020 +1100

    qa/667: fix previous commit
    
    Accidenty pasted some comment text near the start of the file that
    was not supposed to be there.

commit 5983942866240f0b48690b44a072294d6a0406a9
Author: Ken McDonell <kenj@kenj.id.au>
Date:   Thu Jan 23 15:48:47 2020 +1100

    qa/667: various improvements
    
    1. move the pickled/text port from 2004/2003 to 42004/42003 to avoid
       collisions with carbon-cache on Ubunutu (the test checks for this
       but was _notrun for no other good reason)
    2. better filtering of the socat output to avoid buffering and
       ordering problems that were picking the wrong lines although
       the expected lines were also present ... repeated failures on
       RHEL 6.10 (vm28) have now gone away

commit 2a8d714ced93b069c7aa0eeb7815ba40b580a276
Author: Ken McDonell <kenj@kenj.id.au>
Date:   Thu Jan 23 14:57:53 2020 +1100

    qa/check.callback.sample: make sure pmlogger failures are reported
    
    Logic error failed to report details when test 870 fails twice and
    output is the same both times.

commit 47b39cf3586d66a9ad9502bfaa9b022498cc2b82
Author: Ken McDonell <kenj@kenj.id.au>
Date:   Thu Jan 23 06:49:15 2020 +1100

    qa/admin/other-packages: changes for Ubuntu 19.10

commit 5d0d9dcc9597f67b6b23516cf11a2d8cb3b6a0d8
Author: Ken McDonell <kenj@kenj.id.au>
Date:   Wed Jan 22 11:53:31 2020 +1100

    qa/1695: increase delay so valgrind can get started
    
    On some VMs this takes a _long_ time, and there is no reliable
    way to know when it is safe to start the test.

commit cc45581faee8bcbf37d0b8a3e9515f5401916417
Author: Ken McDonell <kenj@kenj.id.au>
Date:   Wed Jan 22 11:46:19 2020 +1100

    qa/510: try to fix failures on RHEL/CentOS 6.10
    
    This has been a problematic test for vm14 and vm28 for sometime.
    
    The change takes a much more anal approach to stopping and restarting
    the primary pmlogger instance and seems to be more robust.

commit c44cc9acbb12d7116e45c7c0fbc171c98a7a8d21
Author: Ken McDonell <kenj@kenj.id.au>
Date:   Tue Jan 14 16:10:40 2020 +1100

    qa/1211: drop QAHOST filtering ... not needed
    
    And was causing a a failure on bozo where bozo-laptop -> QAHOST-laptop

commit 6463d0f76ae9dbb8205a1687c1fdd1d6607bda8b
Author: Ken McDonell <kenj@kenj.id.au>
Date:   Tue Jan 14 16:06:04 2020 +1100

    qa/1155,11157,1170 and 1178: _not run for Ubuntu 19.10
    
    BPF compiler and linux headers are not getting along.

commit 02e6e18a916a1c98a6f0f6633f8943d9f45dc006
Author: Ken McDonell <kenj@kenj.id.au>
Date:   Fri Jan 10 06:49:53 2020 +1100

    qa/054: try to avoid "hang" in QA Farm
    
    The (added) comment says it all ...
    
        This used to log just hinv.ncpu, but as of late 2019 I started seeing
        this test hanging and the QA runs never finished across a number of
        machines in the QA Farm.  The problem is the remote pmcd (usually, but
        not always, vm01 for me), where the pmDesc for hinv.ncpu was unavailable
        because there was no PMCD agent for the domain ... this means the linux
        PMDA was dead, but pmcd was alive.
        Added sample.seconds in the hope that this will avoid the problem for
        this test ... leaving the linux PMDA issue to be investigated separately.

commit 0a1883769a28654ac63609ade9d805d296598114
Author: Ken McDonell <kenj@kenj.id.au>
Date:   Fri Jan 10 06:40:23 2020 +1100

    qa/1423: extend sleeps
    
    There is a risky and racy timing issue in this test.
    
    On vm01 it left QA hung as pcp2spark was in a state from which it
    would never exit.
